auditlog/audit_log_record.go

```diff
@@ -0,0 +1,100 @@
+package auditlog
+
+import (
+	"database/sql"
+	"database/sql/driver"
+	"encoding/json"
+	"reflect"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/fabric8-services/fabric8-common/convert"
+
+	uuid "github.com/satori/go.uuid"
+)
+
+// Record an audit log record to track usage of the service
+type Record struct {
+	ID          uuid.UUID   `sql:"type:uuid default uuid_generate_v4()" gorm:"primary_key"`
+	CreatedAt   time.Time   `json:"created_at,omitempty"`
+	IdentityID  uuid.UUID   `sql:"type:uuid"` // TODO: or should we store the sub claim instead as a string: no need to convert it into an UUID each time
+	EventTypeID uuid.UUID   `sql:"type:uuid"`
+	EventParams EventParams `sql:"type:jsonb"`
+}
+
+const (
+	recordTableName = "audit_logs"
```

Comment:
Could we just call the table `audit_log`

Reply:
I have to admit that I prefer the plural form for tables, as we've been doing in `fabric8-wit` (but I know that it's not exactly the case with the newest tables in `fabric8-auth`)

Reply:
on second thought and after a bit of research, I agree that the singular form for naming tables is more suitable, so I'll update both the `audit_log` and `event_type` table names in the schema. Thanks for pointing this to me, @sbryzak !

Reply:
done in 111ae7eb7d9978c56dbf97c9e2448578d23e18b6

Reply:
Yeah, we adopted that pretty late. A lot of our initial tables are plural.